src/helics/core/ActionMessage.cpp

```diff
@@ -405,7 +417,7 @@ int ActionMessage::fromByteArray(const char* data, int buffer_size)
         payload.assign(data, sz);
         data += sz;
     }
-    int stringCount = *data;
+    int stringCount = static_cast<unsigned char>(*data);
```

Comment:
There's a max of 255 strings? Should this be a `uint8_t`?

Reply:
Yes there is a max of 255 strings.  And the variable needs to be an int for later comparison and use.  So either it gets cast here or later,   clang tidy complained about *data being a signed char so the recommendation was to cast to an unsigned char and then use the implicit conversion to int(which C++ does to everything anyways).